### DISCUSSION TITLE
What are all the meta-data file types and locations (aka .github dir)

### DISCUSSION THREAD

**Discussion by brianddk** [community/community#22144] [New to GitHub]
<p>So I think I know all the GIT meta-data file types, stolen from <a href="https://stackoverflow.com/a/12719247" title="stackoverflow" rel="nofollow noopener">stackoverflow</a>:</p>
<ul>
<li>
<strong><a href="http://git-scm.com/docs/gitignore" target="_blank" rel="nofollow noopener">.gitignore</a></strong> - List of blobs for git to ignore. Affects commands like git add and git clean.</li>
<li>
<strong><a href="http://git-scm.com/docs/gitattributes" target="_blank" rel="nofollow noopener">.gitattributes</a></strong> - Let’s you define attributes on files (e.g., to change how files look in a diff).</li>
<li>
<strong><a href="http://git-scm.com/docs/git-shortlog" target="_blank" rel="nofollow noopener">.mailmap</a></strong> - Lets you tell git that duplicate names or emails in the history are actually the same person. Affects commmands like git shortlog -ns, or git log --format="%aN &lt;%aE&gt;".</li>
<li>
<strong><a href="http://www.kernel.org/pub/software/scm/git/docs/gitmodules.html" target="_blank" rel="nofollow noopener">.gitmodules</a></strong>  - Let’s you define submodules (subdirectories of your git repository which are checkouts of other git repositories).</li>
<li>
<strong>*.keep</strong>  - Something to do with making git gc ignore packs. I couldn’t find much info on this, so I’m not even sure if it’s a file that you commit to the repository or just something that lives in .git. </li>
</ul>
<p>But what are the github specific meta files.  All I know of right now are:</p>
<ul>
<li>
<strong>.github/README.md</strong> - The default readme rendered on your main repo page</li>
<li>
<strong>.github/ISSUE_TEMPLATE</strong>  - The template use for new issue creation.</li>
<li>
<strong>docs/</strong>  - Folder used to render Github Pages jekyll matter.</li>
</ul>
<p>Any others that anyone could add, or correct?</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/brianddk">@brianddk</a>! :wave:</p>
<p>This is an interesting list to put together, thank you!</p>
<p>There are pull request templates to go alongside the issue templates you already mention:</p>
<p><a href="https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository" class="onebox" target="_blank">https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository</a></p>
<p>You could probably add the various community health files such as <strong>CODE_OF_CONDUCT</strong>, <strong>CONTRIBUTING</strong>, and <strong>LICENSE</strong> to that list. They are detailed here:</p>
<p><a href="https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file" class="onebox" target="_blank">https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file</a></p>
<p>The <strong>CODEOWNERS</strong> file might be another good addition:</p>
<p><a href="https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners" class="onebox" target="_blank">https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners</a></p>
<p>Possibly <strong>dependabot.yml</strong> too?</p>
<p><a href="https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates" class="onebox" target="_blank">https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates</a></p>
<p>The <strong>action.yml</strong> of GiHub Actions?</p>
<p><a href="https://help.github.com/en/actions/creating-actions/about-actions" class="onebox" target="_blank">https://help.github.com/en/actions/creating-actions/about-actions</a></p>
<p>I wonder if there are more I’m forgetting about? :thinking:</p>
